pytorch_lightning/callbacks/pt_callbacks.py

```diff
@@ -193,17 +196,35 @@ def __init__(self, filepath, monitor='val_loss', verbose=0,
 
         if mode == 'min':
             self.monitor_op = np.less
-            self.best = np.Inf
+            self.kth = np.Inf
+            self.mode = 'min'
         elif mode == 'max':
             self.monitor_op = np.greater
-            self.best = -np.Inf
+            self.kth = -np.Inf
+            self.mode = 'max'
         else:
             if 'acc' in self.monitor or self.monitor.startswith('fmeasure'):
                 self.monitor_op = np.greater
-                self.best = -np.Inf
+                self.kth = -np.Inf
+                self.mode = 'max'
             else:
                 self.monitor_op = np.less
-                self.best = np.Inf
+                self.kth = np.Inf
+                self.mode = 'min'
+
+    def del_model(self, filepath):
+        dirpath = '/'.join(filepath.split('/')[:-1])
+
+        # make paths
+        os.makedirs(os.path.dirname(filepath), exist_ok=True)
+
+        for filename in os.listdir(dirpath):
+            if self.prefix in filename:
+                path_to_delete = os.path.join(dirpath, filename)
+                try:
```

Comment:
this should be overrun by setting `ignore_erros=True` so 
```python
shutil.rmtree(path_to_delete, ignore_erros=True)
```

Reply:
I'm not sure if this should be changed to `ignore_errors` and `onerror` https://docs.python.org/3/library/shutil.html#shutil.rmtree

This part of the code comes from the original `save_model`

Reply:
I think that's the parameter is missing in Py2, that's why...

Reply:
I think that's the parameter is missing in Py2, that's why...

Reply:
I remember py2 is not supported according to readme

Reply:
@Ir1d yes, that's why I proposed this kind of simplification...